engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -83,27 +85,9 @@
   // If there are errors processing tuples, don't log an error for every tuple as it could overwhelm the logs.
   // The property specifies the minumum number of tuples between two consecutive error log statements. Set it to zero to
   // log every tuple error
-  private static long ERROR_LOG_GAP;
+  private static long ERROR_LOG_GAP = SystemPropertyReader.getLong("org.apache.apex.stram.tupleRecorder.errorLogGap", 10000L, 0, Integer.MAX_VALUE);
```

Comment:
Because of using maxValue of Integer.MAX_VALUE the behavior now has changed. The old code only disallowed  values < 0 but with this even values > Integer.MAX_VALUE are disallowed. It may be intentional and a good check to have (but I doubt that is the case for the error-log-gap number)